xarray/core/accessor_str.py

```diff
@@ -865,49 +1061,838 @@ def replace(self, pat, repl, n=-1, case=None, flags=0, regex=True):
+        ...         ["test0\\ntest1\\ntest2\\n\\ntest3", "", "abra  ka\\nda\\tbra"],
+        ...     ],
+        ...     dims=["X", "Y"],
+        ... )
+
+        Split once and put the results in a new dimension
+
+        >>> values.str.split(dim="ZZ", maxsplit=1)
+        <xarray.DataArray (X: 2, Y: 3, ZZ: 2)>
+        array([[['abc', 'def'],
+                ['spam', 'eggs\\tswallow'],
+                ['red_blue', '']],
+        <BLANKLINE>
+               [['test0', 'test1\\ntest2\\n\\ntest3'],
+                ['', ''],
+                ['abra', 'ka\\nda\\tbra']]], dtype='<U18')
+        Dimensions without coordinates: X, Y, ZZ
+
+        Split as many times as needed and put the results in a new dimension
+
+        >>> values.str.split(dim="ZZ")
+        <xarray.DataArray (X: 2, Y: 3, ZZ: 4)>
+        array([[['abc', 'def', '', ''],
+                ['spam', 'eggs', 'swallow', ''],
+                ['red_blue', '', '', '']],
+        <BLANKLINE>
+               [['test0', 'test1', 'test2', 'test3'],
+                ['', '', '', ''],
+                ['abra', 'ka', 'da', 'bra']]], dtype='<U8')
+        Dimensions without coordinates: X, Y, ZZ
+
+        Split once and put the results in lists
+
+        >>> values.str.split(dim=None, maxsplit=1)
+        <xarray.DataArray (X: 2, Y: 3)>
+        array([[list(['abc', 'def']), list(['spam', 'eggs\\tswallow']),
+                list(['red_blue'])],
+               [list(['test0', 'test1\\ntest2\\n\\ntest3']), list([]),
+                list(['abra', 'ka\\nda\\tbra'])]], dtype=object)
+        Dimensions without coordinates: X, Y
+
+        Split as many times as needed and put the results in a list
+
+        >>> values.str.split(dim=None)
+        <xarray.DataArray (X: 2, Y: 3)>
+        array([[list(['abc', 'def']), list(['spam', 'eggs', 'swallow']),
+                list(['red_blue'])],
+               [list(['test0', 'test1', 'test2', 'test3']), list([]),
+                list(['abra', 'ka', 'da', 'bra'])]], dtype=object)
+        Dimensions without coordinates: X, Y
+
+        Split only on spaces
+
+        >>> values.str.split(dim="ZZ", sep=" ")
+        <xarray.DataArray (X: 2, Y: 3, ZZ: 3)>
+        array([[['abc', 'def', ''],
+                ['spam\\t\\teggs\\tswallow', '', ''],
+                ['red_blue', '', '']],
+        <BLANKLINE>
+               [['test0\\ntest1\\ntest2\\n\\ntest3', '', ''],
+                ['', '', ''],
+                ['abra', '', 'ka\\nda\\tbra']]], dtype='<U24')
+        Dimensions without coordinates: X, Y, ZZ
+
+        See Also
+        --------
+        DataArray.str.rsplit
+        str.split
+        pandas.Series.str.split
+        """
+        if sep is not None:
+            sep = self._obj.dtype.type(sep)
+
+        if dim is None:
+            f = lambda x: x.split(sep, maxsplit)
+            return self._apply(f, dtype=np.object_)
+
+        if not self._obj.size:
+            return self._obj.copy().expand_dims({dim: 0}, -1)
```

Comment:
What are you doing here?

Reply:
Working around an issue in `_apply`.  I added a comment explaining this.